presto-native-execution/src/test/java/com/facebook/presto/nativeworker/NativeQueryRunnerUtils.java

```diff
@@ -360,6 +362,21 @@ public static void createRegion(QueryRunner queryRunner)
         }
     }
 
+    public static void createTableToTestHiddenColumns(QueryRunner queryRunner)
+    {
+        if (!queryRunner.tableExists(queryRunner.getDefaultSession(), "test_hidden_columns")) {
+            queryRunner.execute("CREATE TABLE test_hidden_columns (regionkey bigint, name varchar(25), comment varchar(152))");
+
+            // Inserting two rows with 2 seconds delay to have a different modified timestamp for each file.
+            queryRunner.execute("INSERT INTO test_hidden_columns SELECT * FROM region where regionkey = 0");
+            try {
+                TimeUnit.SECONDS.sleep(2);
```

Comment:
@nmahadevuni This looks a bit confusing. I am seeing tests failing with errors 

```
[INFO] -------------------------------------------------------
[INFO]  T E S T S
[INFO] -------------------------------------------------------
[INFO] Running com.facebook.presto.facebook.nativeworker.oss.TestPrestoNativeGeneralQueriesThrift
[ERROR] Tests run: 61, Failures: 2, Errors: 0, Skipped: 0, Time elapsed: 191.177 s <<< FAILURE! - in com.facebook.presto.facebook.nativeworker.oss.TestPrestoNativeGeneralQueriesThrift
[ERROR] com.facebook.presto.facebook.nativeworker.oss.TestPrestoNativeGeneralQueriesThrift.testFileModifiedTimeHiddenColumn  Time elapsed: 0.257 s  <<< FAILURE!
java.lang.AssertionError: expected [1] but found [2]
	at org.testng.Assert.fail(Assert.java:110)
	at org.testng.Assert.failNotEquals(Assert.java:1413)
	at org.testng.Assert.assertEqualsImpl(Assert.java:149)
	at org.testng.Assert.assertEquals(Assert.java:131)
	at org.testng.Assert.assertEquals(Assert.java:954)
	at com.facebook.presto.nativeworker.AbstractTestNativeGeneralQueries.testFileModifiedTimeHiddenColumn(AbstractTestNativeGeneralQueries.java:1069)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.testng.internal.invokers.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:135)
	at org.testng.internal.invokers.TestInvoker.invokeMethod(TestInvoker.java:673)
	at org.testng.internal.invokers.TestInvoker.invokeTestMethod(TestInvoker.java:220)
	at org.testng.internal.invokers.MethodRunner.runInSequence(MethodRunner.java:50)
	at org.testng.internal.invokers.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:945)
	at org.testng.internal.invokers.TestInvoker.invokeTestMethods(TestInvoker.java:193)
	at org.testng.internal.invokers.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:146)
	at org.testng.internal.invokers.TestMethodWorker.run(TestMethodWorker.java:128)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:829)

[ERROR] com.facebook.presto.facebook.nativeworker.oss.TestPrestoNativeGeneralQueriesThrift.testFileSizeHiddenColumn  Time elapsed: 0.693 s  <<< FAILURE!
java.lang.AssertionError: expected [1] but found [2]
	at org.testng.Assert.fail(Assert.java:110)
	at org.testng.Assert.failNotEquals(Assert.java:1413)
	at org.testng.Assert.assertEqualsImpl(Assert.java:149)
	at org.testng.Assert.assertEquals(Assert.java:131)
	at org.testng.Assert.assertEquals(Assert.java:954)
	at com.facebook.presto.nativeworker.AbstractTestNativeGeneralQueries.testFileSizeHiddenColumn(AbstractTestNativeGeneralQueries.java:1054)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.testng.internal.invokers.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:135)
	at org.testng.internal.invokers.TestInvoker.invokeMethod(TestInvoker.java:673)
	at org.testng.internal.invokers.TestInvoker.invokeTestMethod(TestInvoker.java:220)
	at org.testng.internal.invokers.MethodRunner.runInSequence(MethodRunner.java:50)
	at org.testng.internal.invokers.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:945)
	at org.testng.internal.invokers.TestInvoker.invokeTestMethods(TestInvoker.java:193)
	at org.testng.internal.invokers.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:146)
	at org.testng.internal.invokers.TestMethodWorker.run(TestMethodWorker.java:128)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:829)

[INFO] 
[INFO] Results:
[INFO] 
[ERROR] Failures: 
[ERROR]   TestPrestoNativeGeneralQueriesThrift>AbstractTestNativeGeneralQueries.testFileModifiedTimeHiddenColumn:1069 expected [1] but found [2]
[ERROR]   TestPrestoNativeGeneralQueriesThrift>AbstractTestNativeGeneralQueries.testFileSizeHiddenColumn:1054 expected [1] but found [2]
[INFO] 
[ERROR] Tests run: 61, Failures: 2, Errors: 0, Skipped: 0
```

Any thoughts what can go wrong?

CC: @xiaoxmeng

Reply:
This is not specific to any format. I have tested this locally and verified that the file sizes generated are different for these two inserts and with the delay, the file modified timestamp is also different.

Reply:
I assume this doesn't need any change at worker side? Is there any flakiness in this test? Thanks!

Reply:
No, this doesn't have any changes to be made in the worker side. I have checked that the test is not flaky. I have run it multiple times even today. The native check failure is unrelated.

Reply:
The only way I see these can fail is if the source table region somehow ends with duplicate rows. And when we create test_hidden_columns tables, with each insert, we get two rows rather than the expected one row. Otherwise, I don't see how even the file_modified_timestamp can be same with even 2 seconds delay between the inserts. Can we get the files from the table location when this failure occurs? I can confirm by looking at the file stat info and the number of rows in each file. Or output of `ls -lrth` from the table location.

Reply:
After I run the test in my Mac, the contents of the table are below.

`..tpch/test_hidden_columns % ls -lrth
total 16
-rw-r--r--  1 nmahadevuni  wheel   701B Aug 15 12:10 20240815_064042_00002_79tnd_d18c9b52-ba08-4abe-9e6e-e7d4b45810ad
-rw-r--r--  1 nmahadevuni  wheel   756B Aug 15 12:10 20240815_064054_00003_79tnd_ab091114-d5c4-47e7-9187-782d73e6e0be
nmahadevuni::tpch/test_hidden_columns % stat 20240815_064042_00002_79tnd_d18c9b52-ba08-4abe-9e6e-e7d4b45810ad
16777232 84182810 -rw-r--r-- 1 nmahadevuni wheel 0 701 "Aug 15 12:10:52 2024" "Aug 15 12:10:52 2024" "Aug 15 12:10:52 2024" "Aug 15 12:10:51 2024" 4096 8 0 20240815_064042_00002_79tnd_d18c9b52-ba08-4abe-9e6e-e7d4b45810ad
nmahadevuni::tpch/test_hidden_columns % stat 20240815_064054_00003_79tnd_ab091114-d5c4-47e7-9187-782d73e6e0be
16777232 84182823 -rw-r--r-- 1 nmahadevuni wheel 0 756 "Aug 15 12:10:57 2024" "Aug 15 12:10:56 2024" "Aug 15 12:10:57 2024" "Aug 15 12:10:56 2024" 4096 8 0 20240815_064054_00003_79tnd_ab091114-d5c4-47e7-9187-782d73e6e0be`

Reply:
> can confirm by looking at the file stat info and the number of rows in each file. Or output of `ls -lrth` from the table location.

@nmahadevuni  so the filtering happens at split generation at coordinator side. The worker doesn't seem handle the filer on SYNTHESIZED column. Without this change, the filter works for path but not file size, and I haven't tried the other info column. This PR fixes the filter for other info or hidden columns like file size? Maybe it is due to some test flakiness. @spershin found something flakiness in the test and is going to apply a followup fix on the test.

Reply:
@xiaoxmeng : The synthesized columns were done with https://github.com/prestodb/presto/pull/21965 and https://github.com/facebookincubator/velox/pull/8800